crates/holochain_zome_types/src/timestamp.rs

```diff
@@ -340,18 +356,23 @@ impl Timestamp {
         ts.as_millis().clamp(0, i64::MAX as u128) as i64
     }
 
-    /// Construct a Timestsamp from countersigning session data.
+    /// Construct from milliseconds
+    pub fn from_millis(millis: i64) -> Self {
+        let secs = millis / 1_000;
+        let nanos = (millis % 1_000) as u32 * 1_000_000;
+        Self(secs, nanos)
+    }
+
+    /// Construct a Timestamp from countersigning session data.
     /// Ostensibly used for the Header because the session itself covers a time range.
     pub fn from_countersigning_data(session_data: &CounterSigningSessionData) -> Self {
+        let start = session_data.preflight_request().session_times().start();
         Self(
-            session_data
-                .preflight_request()
-                .session_times()
-                .start()
+            start
                 .0
                 .checked_add(SESSION_HEADER_TIME_OFFSET_MILLIS)
                 .unwrap_or(i64::MAX),
```

Comment:
Hey, isn't this wrong? We're adding milliseconds to seconds. This isn't the only place this happens. This is part of why I made this field private and put it behind a named accessor.

Reply:
Looks wrong to me